test/unit/Swaps.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { expect } from 'chai';
+import Swaps, { SwapDeal } from '../../lib/swaps/Swaps';
+import Logger, { Level } from '../../lib/Logger';
+import { SwapPhase, SwapState, SwapRole } from '../../lib/types/enums';
+
+const loggers = Logger.createLoggers(Level.Warn);
+
+describe('Swaps', () => {
+  const takerDeal: SwapDeal = {
+    role: SwapRole.Taker,
+    phase: SwapPhase.SwapCreated,
+    state: SwapState.Active,
+    peerPubKey: '03029c6a4d80c91da9e40529ec41c93b17cc9d7956b59c7d8334b0318d4a86aef8',
+    orderId: 'f8a85c66-7e73-43cd-9ac4-176ff4cc28a8',
+    localOrderId: '1',
+    proposedQuantity: 0.01,
+    quantity: 0.01,
+    pairId: 'LTC/BTC',
+    price: 0.005,
+    takerCurrency: 'LTC',
+    makerCurrency: 'BTC',
+    takerAmount: 1000000,
+    makerAmount: 5000,
+    takerCltvDelta: 144,
+    r_hash: '62c8bbef4587cff4286246e63044dc3e454b5693fb5ebd0171b7e58644bfafe2',
+    createTime: 1540716251106,
+  };
+
+  const makerDeal = {
+    ...takerDeal,
+    role: SwapRole.Maker,
+    takerCurrency: takerDeal.makerCurrency,
+    makerCurrency: takerDeal.takerCurrency,
+    takerAmount: takerDeal.makerAmount,
+    makerAmount: takerDeal.takerAmount,
```

Comment:
why the currency/amount is being assigned to the opposite side (taker on maker, maker on taker)? I don't see how that's aligned with `handleSwapRequest`, which assigns these values as-is (in `acceptDeal`).

Reply:
What I was going for here is not exactly the same as the flow that happens within `xud`, where two sides negotiate on the details of a single swap - and where the same node is the "taker" on both sides of the deal. Here I was trying to create two *separate* swaps that are mirrors of each other when it comes to taker/maker and which copy other details for convenience. Let me know if that makes sense, and if you think it's worth adding a comment to the tests to try to clarify that.

Reply:
Yes that wasn't clear to me. I assumed that it's just the other side, because they usually go together. I think that the fact that we don't have calculations on the maker side is confusing for this matter. We probably need to fix that before continuing with the tests.

Reply:
Yup there's a TODO for the calculations on the maker side, it's on my radar.